flexmeasures/cli/data_add.py

```diff
@@ -176,7 +190,46 @@ def new_account(name: str, roles: str, consultancy_account: Account | None):
     if account is not None:
         click.secho(f"Account '{name}' already exists.", **MsgStyle.ERROR)
         raise click.Abort()
-    account = Account(name=name, consultancy_account=consultancy_account)
+
+    # Validate color format ---------------------------------------------------
+    # Regular expression for valid hex color code
+    hex_color_pattern = re.compile(r"^#?([A-Fa-f0-9]{6}|[A-Fa-f0-9]{3})$")
+
+    if primary_color and not hex_color_pattern.match(primary_color):
+        click.secho("Primary color is not in valid hex format", **MsgStyle.ERROR)
+        raise click.Abort()
+
+    if secondary_color and not hex_color_pattern.match(secondary_color):
+        click.secho("Secondary color is not in valid hex format", **MsgStyle.ERROR)
+        raise click.Abort()
+
+    # make sure both colors or non are given
+    if primary_color and not secondary_color:
+        click.secho("Secondary color is missing", **MsgStyle.ERROR)
+        raise click.Abort()
+
+    if secondary_color and not primary_color:
+        click.secho("Primary color is missing", **MsgStyle.ERROR)
```

Comment:
, but secondary is given. Add both or none.

Reply:
This has no been refactored 👍